src/AElf.Kernel.Core/Blockchain/Events/TransactionResourceExtractionEvents.cs

```diff
@@ -0,0 +1,15 @@
+using System.Collections.Generic;
+using AElf.Types;
+
+namespace AElf.Kernel.Blockchain.Events
+{
+    public class TransactionResourcesNeededEvent
```

Comment:
I think it's better we use more generic event types instead of creating specific events. Core shouldn't have knowledge about parallel execution (which is an extension).

Reply:
I pushed a new one to use existing events and remove the new ones specific to resource extraction.